sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor.cs

```diff
@@ -781,6 +782,41 @@ async Task performProcessing()
         ///
         protected abstract Task<IEnumerable<EventProcessorCheckpoint>> ListCheckpointsAsync(CancellationToken cancellationToken);
 
+        /// <summary>
+        ///   Returns a checkpoint for the Event Hub, consumer group, and partition ID associated with the
+        ///   event processor instance, so that processing for a given partition can be properly initialized.
+        ///   The default implementation calls the <see cref="ListCheckpointsAsync"/> and filters results by <see cref="EventProcessorCheckpoint.PartitionId"/>.
+        ///   It's recommended that this method is overriden in <see cref="EventProcessor{TPartition}"/> implementations to achieve an optimal performance.
+        /// </summary>
+        ///
+        /// <param name="partitionId">The ID of the partition for which to retrieve the checkpoint.</param>
```

Comment:
the nittiest of nits:  The rest of the public documentation avoids the `ID` abbreviation.

```suggestion
        /// <param name="partitionId">The identifier of the partition for which to retrieve the checkpoint.</param>
```